### DISCUSSION TITLE
Hey,i am a cpp developers i want to make game without sfml

### DISCUSSION THREAD

**Discussion by 639363** [community/community#114556] [Programming Help] [answered]
### Body

My first question is about coordinate how to fix ,change coordinate in cpp 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mairinkdev** [comment, marked as answer]
**If you're not using SFML, you'll need to handle coordinates manually. In C++, you can define your own coordinate system using variables or structures to represent points or vectors. For example, you could define a Point struct:**

```
struct Point {
     int x; 
     int y;
};
```

**Then, you can create instances of this struct to represent coordinates:**

```
Point p1 = {10, 20};
```

**To change coordinates, you simply modify the values of the x and y members of the struct:**

```
p1. x = 30;
p1.у = 40;
```

**This allows you to work with coordinates in your own custom way without relying on external libraries like
SFML.**

**Comment by 639363** [comment]
But how i show position of player for example player let be # how to move # up down left right and thank you for giving information

**Reply by mairinkdev** [reply]
In this C++ code, I've developed a simple game where the player can move a character represented by '#' within a game space defined by an 80 columns by 20 rows grid. To achieve this, I used variables to control the player's position (playerX and playerY). I implemented a function to clear the screen (clearScreen()) and another one to draw the game on the screen (draw()). Keyboard input detection is done using _kbhit() and _getch() from the <conio.h> library, allowing the player to move the character using the 'w', 's', 'a', and 'd' keys. It's a basic solution, but a good starting point for developing a more complex game.

```
#include <iostream>
#include <cstdlib> // For rand() and srand()
#include <ctime>   // For time()
#include <conio.h> // For _getch() and _kbhit()

// Define constants for screen dimensions
const int SCREEN_WIDTH = 80;
const int SCREEN_HEIGHT = 20;

// Define constants for player character and starting position
const char PLAYER_CHAR = '#';
int playerX = SCREEN_WIDTH / 2;
int playerY = SCREEN_HEIGHT / 2;

// Function to clear the screen
void clearScreen() {
    system("cls"); // For Windows
    // system("clear"); // For Unix/Linux
}

// Function to draw the game board
void draw(char* board) {
    // Draw player
    board[playerY * SCREEN_WIDTH + playerX] = PLAYER_CHAR;

    // Draw board
    for (int i = 0; i < SCREEN_HEIGHT; ++i) {
        for (int j = 0; j < SCREEN_WIDTH; ++j) {
            std::cout << board[i * SCREEN_WIDTH + j];
        }
        std::cout << std::endl;
    }
}

int main() {
    // Seed the random number generator
    std::srand(static_cast<unsigned int>(std::time(nullptr)));

    // Initialize the game board
    char board[SCREEN_WIDTH * SCREEN_HEIGHT];
    for (int i = 0; i < SCREEN_WIDTH * SCREEN_HEIGHT; ++i) {
        board[i] = ' ';
    }

    // Game loop
    bool gameOver = false;
    while (!gameOver) {
        // Clear the screen
        clearScreen();

        // Process player input
        if (_kbhit()) {
            char input = _getch();
            switch (input) {
                case 'w':
                    playerY--;
                    break;
                case 's':
                    playerY++;
                    break;
                case 'a':
                    playerX--;
                    break;
                case 'd':
                    playerX++;
                    break;
                case 'q':
                    gameOver = true;
                    break;
                default:
                    break;
            }
        }

        // Draw the game board
        draw(board);

        // Wait for a short time
        std::this_thread::sleep_for(std::chrono::milliseconds(100));
    }

    return 0;
}
```

**Reply by 639363** [reply]
Can you explain why you using char*
And what is :
srand(static_cast<unsigned int>(time(nullptr)); 
Can you explain draw function one by one because
I am an beginner

**Comment by 639363** [comment]
Can you explain why you using char*
And what is :
``srand(static_cast<unsigned int>(time(nullptr));
``
Can you explain draw function one by one because
I am an beginner
